packages/driver/src/cy/commands/navigation.ts

```diff
@@ -916,33 +1026,47 @@ export default (Commands, Cypress, cy, state, config) => {
         const existingHash = remote.hash || ''
         const existingAuth = remote.auth || ''
 
-        if (previousDomainVisited && (remote.originPolicy !== existing.originPolicy)) {
+        if (previousUrlVisited && (remote.originPolicy !== existing.originPolicy)) {
```

Comment:
previousUrlVisited is no longer a boolean but a location obj. Should there be additional checks to verify this is indeed a previously visted url?

Reply:
I think that's already covered by the `remote.originPolicy !== existing.originPolicy` check here, but I may be misunderstanding. Why would we need an additional check?

Reply:
fyi, `previousUrlVisited` was incorrectly typed as a boolean before and was always a location obj.